### PR TITLE
Testing plan for location verification

### LINKED ISSUE #85: Location Verification Implementation Guidelines
**Problem description**
There is no clear guideline about how to implement the response to location-verification, when result is PARTIAL, and it is not obvious how to calculate the matchRate 

**Expected action**
Trigger discussion and agree on some guidelines for common implementation.

An initial proposal is presented, to cover several scenarios that may happen. Also with comments and concerns.

**Additional context**

Word Document:
[Location Verification Implementation Guidelines.docx](https://github.com/camaraproject/DeviceLocation/files/12213109/Location.Verification.Implementation.Guidelines.docx)

### LINKED ISSUE #208: Test plan for Location Verification API
**Problem description**
Test plan for Location Verification API

**Expected action**
Provide test plan aligned with [Testing guidelines in Commonalities](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Testing-Guidelines.md)

### PR DESCRIPTION
**Pull request by jlurien**
#### What type of PR is this?

* tests

#### What this PR does / why we need it:

Example of testing plan aligned with the guidelines being proposed in https://github.com/camaraproject/Commonalities/issues/158.

#### Which issue(s) this PR fixes:

Fixes #208 

#### Special notes for reviewers:

There are many comments embedded in the test plan to clarify and explain some scenarios. In some cases alternate syntax for the same functionality is used to illustrate the possibilities.

Some points are still open to discussion, such as the degree of specification of the testing plan.

### CODE REVIEW

**Review by bigludo7 — DISMISSED** [review]
Checked with @patrice-conil and work for Orange side.
Could be very helpful to provide an extract from that to answer https://github.com/camaraproject/Commonalities/issues/127 in commonalities.

**Comment by jlurien** [comment]
> Checked with @patrice-conil and work for Orange side. Could be very helpful to provide an extract from that to answer [camaraproject/Commonalities#127](https://github.com/camaraproject/Commonalities/issues/127) in commonalities.

Yes, I have to think how to align this with the work in Commonalities and the testing guidelines enhancement. I will be preparing some more detailed proposal towards the dedicated workshop.

**Comment by jlurien** [comment]
> @jlurien Accordingly to your proposal in Commonalities, this file name should location-verification.feature I guess.

True, I have to update and align my own example with my own proposal :)

**Review by bigludo7 — DISMISSED** [review]
LGTM for me
2 very minor typing errors

File: code/Test_definitions/location-verification.feature
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:2]:
```suggestion
  # Input to be provided by the implementation to the tester

```

File: code/Test_definitions/location-verification.feature
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:11]:
```suggestion
  # * A device object identifying a device commercialized by the implementation for which the service is not applicable, if any

```

File: code/Test_definitions/location-verification.feature
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:3]:
```suggestion
```

File: code/Test_definitions/location-verification.feature
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:161]:
```suggestion
    And the response property "$.code" is "INVALID_TOKEN_CONTEXT"
```

File: code/Test_definitions/location-verification.feature
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:252]:
```suggestion
    And the response property "$.code" is "AUTHENTICATION_REQUIRED"
```

File: code/Test_definitions/location-verification.feature
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:252]:
```suggestion
    And the response property "$.code" is "UNAUTHENTICATED"
```

**Comment by jlurien** [comment]
@bigludo7 Wee need another explicit approval after the last fixes to merge the PR. Thanks

**Review by bigludo7 — APPROVED** [review]
LGTM

**Pull request merged**
